control_panel_api/models.py

```diff
@@ -78,10 +77,12 @@ def _repo_name(self):
 
         "https://github.com/org/a_repo_name" => "a_repo_name"
         "git@github.com:org/repo_2.git" => "repo_2"
-        '''
-
-        last_after_slash = self.repo_url.split('/')[-1]
-        return last_after_slash.replace('.git', '')
+        """
+        last_after_slash = self.repo_url \
+            .rstrip('/') \
+            .replace('.git', '') \
+            .rsplit('/', 1)[1]
+        return last_after_slash
```

Comment:
This code is not very readable, especially with the line continuation slashes and method chaining.
It would be more readable like this:
```python
url = self.repo_url

if url.endswith('/'):
    url = url[:-1]

if url.endswith('.git'):
    url = url[:-4]

return url.rsplit('/', 1)
```

Reply:
👍